### PR TITLE
Auto-detect Arduinos attached to USB serial ports

### LINKED ISSUE #277: Re-enable auto_detect of Arduinos
The Arduinos aren't always associated with the same /dev/ttyACM device, so the currently disabled auto_detect feature would be valuable.

### PR DESCRIPTION
**Pull request by jamessynge**
Fix #277.

This is only lightly tested. I still need to connect a suitable Arduino to my laptop for proper testing.

### CODE REVIEW

File: peas/sensors.py [hunk 1]
```diff
@@ -25,45 +21,25 @@ def __init__(self, auto_detect=False, *args, **kwargs):
...
         self.db = None
         self.messaging = None
 
         # Store each serial reader
         self.serial_readers = dict()
 
-        if auto_detect:
+        if auto_detect or self.config['environment'].get('auto_detect', False):
+            self.logger.debug('Performing auto-detect')
             for port_num in range(9):
```
Comment (wtgee) [inline peas/sensors.py]: We can also grab connected ports and just check those: https://pythonhosted.org/pyserial/tools.html?highlight=comport#serial.tools.list_ports.comports. We could then check vendor id and product id. The arduino uno even reports a `serial_number` correctly although I don't think the micros do.

That could probably help clean up some of the `_auto_detect_port` below.

File: peas/sensors.py [hunk 2]
```diff
@@ -141,20 +150,20 @@ def capture(self, use_mongo=True, send_message=True):
                 if send_message:
                     self.send_message({'data': data}, channel='environment')
             except yaml.parser.ParserError:
```
Comment (wtgee) [inline peas/sensors.py]: I think I didn't know this when I wrote this a long time ago, but you can pass a tuple for except:

```
except (yaml.parser.ParserError, ValueError, TypeError, Exception) as e:
```

However if we have `Exception` at the end and we are not doing anything different for the different exceptions then really we just need to capture the generic.

File: peas/sensors.py (hunk 1 above)
Comment (jamessynge) [reply peas/sensors.py]: Submitted #284 for this.

File: peas/sensors.py (hunk 2 above)
Comment (jamessynge) [reply peas/sensors.py]: Hmm, I didn't know that either, though as you say it doesn't do much to include both a class and super-class in the the same tuple.

**Pull request merged**